src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -975,12 +1353,89 @@ export class SavedObjectsRepository {
     const savedObject = this._serializer.rawToSavedObject(raw);
     return omit(savedObject, 'namespace');
   }
+
+  private rawDocExistsInNamespace(raw: SavedObjectsRawDoc, namespace?: string) {
+    const rawDocType = raw._source.type as string;
+
+    // if the type is namespace isolated, or namespace agnostic, we can continue to rely on the guarantees
+    // of the document ID format and don't need to check this
+    if (!this._registry.isMultiNamespace(rawDocType)) {
+      return true;
+    }
```

Comment:
Even if this is private and all usages are using this correctly, I find this a little dangerous, as without reading this comment, the name of the function makes think this should work for single namespace objects. Maybe we could either do the check on single namespaced object based on the `id`, rename the function to `rawDocIsNotMultiNamespaceOrExistsInNamespace` (yea, best name ever I know), or at least add function-level jsdoc?

Reply:
Yeah, I understand the concern. Since this is a private method, it is only ever called within the repository -- and it is/should only ever be called on raw documents that have just been fetched from Elasticsearch.

I think the best thing to do here is to add a jsdoc to point out that this should only be used for documents that were retrieved from Elasticsearch.